src/NHibernate.Test/NHSpecificTest/NH3023/DeadlockHelper.cs

```diff
@@ -27,7 +29,7 @@ public void ForceDeadlockOnConnection(SqlConnection connection)
 						{
 							using (var scope = new TransactionScope(TransactionScopeOption.RequiresNew))
 							{
-								using (var cxn = new SqlConnection(connection.ConnectionString + ";Pooling=No"))
+								using (var cxn = new SqlConnection(connectionString + ";Pooling=No"))
```

Comment:
`connection.ConnectionString` is purged from password, when there is one. This cause the winner thread to fail in such case.